mlonmcu/target/riscv/ara.py

```diff
@@ -0,0 +1,300 @@
+#
+# Copyright (c) 2022 TUM Department of Electrical and Computer Engineering.
+#
+# This file is part of MLonMCU.
+# See https://github.com/tum-ei-eda/mlonmcu.git for further info.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""MLonMCU ARA Target definitions"""
+
+import os
+import re
+from pathlib import Path
+
+from mlonmcu.logging import get_logger
+from mlonmcu.feature.features import SUPPORTED_TVM_BACKENDS
+from mlonmcu.target.common import cli, execute
+from mlonmcu.target.metrics import Metrics
+from .riscv import RISCVTarget
+from .util import update_extensions
+
+logger = get_logger()
+
+
+class AraTarget(RISCVTarget):
+    """Target using a Pulpino-like VP running in the GVSOC simulator"""
+
+    FEATURES = RISCVTarget.FEATURES + ["log_instrs"]
+
+    DEFAULTS = {
+        **RISCVTarget.DEFAULTS,
+        "xlen": 64,
+        "abi": "lp64d",
+        "extensions": ["g", "c", "v"],  # TODO overwrite extensions elegantly
+        "nr_lanes": 4,
+        "vlen": 4096,
+    }
+
+    REQUIRED = RISCVTarget.ARA_GCC_TOOLCHAIN_REQUIRED + [
+        "ara.apps_dir",  # for the bsp package usded in compilation
+        "ara.hardware_dir",  # for the rtls
+        "ara.bender_path",  # for simulation
+        "ara.verilator_install_dir",  # for simulation
+        "ara.tb_verilator_build_dir"  # actually just a tmp folder, recommended to be under hardware/build
```

Comment:
This would not work for parallel runs. Please use https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory instead.